src/dbnode/storage/bootstrap/bootstrapper/fs/README.md

```diff
@@ -0,0 +1,28 @@
+# fs bootstrapper
+
+The fs bootstrapper inspects both data and index info files on disk and marks requested shard time ranges
+for each as fulfilled respectively when it finds persisted data for either.
+
+The fs bootstrapper *only* bootstraps from persisted data (both index and TSDB) on disk. It passes along unfulfilled
+shard time ranges to the next bootstrapper (as does every bootstrapper).
+
+## TSDB data on disk missing index data
+
+There are a few cases where TSDB blocks may be missing an index block on disk.
+  1. TSDB blocks smaller than index blocks
+    - TSDB blocks can exist on disk that don't cover the entire index block
+  2. Node crash between a succesful warm TSDB flush(es) (for the entire index block)
+     and successful index flush.
+  3. Crash after streaming TSDB blocks from peer during peers bootstrapping.
+    - Shard will still be marked as initializing.
+
+We handle case 1 by relying on bootstrapping from index snapshots/commitlogs to load in-mem index data.
+We handle case 2 by just waiting for a successful index warm flush to complete.
+We handle case 3 by passing unfulfilled index shard time ranges along to the peers bootstrapper (for uninitialized shards) so that it can build index segments for TSDB blocks missing index data.
```

Comment:
@robskillington I think this section is particularly important. I documented the expectations for handling of diff cases where we previously would have built an index segment in the fs bootstrapper and how its handled now since we removed index segment building from the fs bootstrapper.

Want to make sure we're covering all our bases here.

Reply:
+1

Can we move this to do `src/docs/bootstrapping.md` as per other suggestions? Maybe a new section called `# Filesystem bootstrapping` in there?